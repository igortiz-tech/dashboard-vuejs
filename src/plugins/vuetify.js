import { createVuetify } from 'vuetify'
import { aliases, mdi } from 'vuetify/iconsets/mdi'

export default createVuetify({
    icons: {
        defaultSet: 'mdi',
        aliases,
        sets: {
            mdi,
        }
    },
    theme: {
        themes: {
            light: {
                primary: "#1887B4",
                secondary: "#673ab7",
                accent: "#3f51b5",
                error: "#f44336",
                warning: "#ffc107",
                info: "#00bcd4",
                success: "#4caf50",
            },
        },
        options: {
            customProperties: true,
        },
    },
})
