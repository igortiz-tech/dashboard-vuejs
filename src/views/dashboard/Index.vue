<template>

  <div
      class="d-flex flex-row"
  >
    <v-card class="ma-2 pa-2" width="400" elevation="0">
      <template v-slot:title>
        This is a title
      </template>

      <template v-slot:subtitle>
        This is a subtitle
      </template>

      <template v-slot:text>
        This is content
      </template>
    </v-card>
    <v-card class="ma-2 pa-2" width="400" elevation="0">
      <template v-slot:title>
        This is a title
      </template>

      <template v-slot:subtitle>
        This is a subtitle
      </template>

      <template v-slot:text>
        This is content
      </template>
    </v-card>
  </div>

  <div
      class="d-flex flex-row"
  >
    <v-card class="ma-2 pa-2" width="400" elevation="0">
      <template v-slot:title>
        This is a title
      </template>

      <template v-slot:subtitle>
        This is a subtitle
      </template>

      <template v-slot:text>
        This is content
      </template>
    </v-card>
  </div>

</template>

<script>
export default {
  name: "DasboardIndex"
}
</script>

<style scoped>

</style>