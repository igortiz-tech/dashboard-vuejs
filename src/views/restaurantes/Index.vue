<template>
    <div id="Restaurante">
      <v-container fluid class="dataTable" >
        <v-row>
          <v-col>
            <material-card-flat
                color="blue"
                title="Listagem de Restaurante"
                text="Utilize os filtros para facilitar a busca por restaurante."
            >
              <div>
                <v-table density="compact">
                <thead>
                <tr>
                  <th class="text-left">
                    ID
                  </th>
                  <th class="text-left">
                    Nome
                  </th>
                  <th class="text-left">
                  </th>
                </tr>
                </thead>
                <tbody>
                <tr
                    v-for="item in restaurantes"
                    :key="item.id"
                >
                  <td>{{ item.id }}</td>
                  <td>{{ item.nome }}</td>
                  <td class="text-right">
                    <v-icon icon="mdi-dots-vertical"></v-icon>
                  </td>
                </tr>
                </tbody>
              </v-table>
              </div>
            </material-card-flat>
          </v-col>
        </v-row>
      </v-container>

      <v-container fluid class="dataTable" >
        <v-row>
          <v-col>
            <material-card-flat
                color="blue"
                title="Listagem de Restaurante"
                text="Utilize os filtros para facilitar a busca por restaurante."
            >
              <div>
                <v-table density="compact">
                  <thead>
                  <tr>
                    <th class="text-left">
                      ID
                    </th>
                    <th class="text-left">
                      Nome
                    </th>
                    <th class="text-left">
                    </th>
                  </tr>
                  </thead>
                  <tbody>
                  <tr
                      v-for="item in restaurantes"
                      :key="item.id"
                  >
                    <td>{{ item.id }}</td>
                    <td>{{ item.nome }}</td>
                    <td class="text-right">
                      <v-icon icon="mdi-dots-vertical"></v-icon>
                    </td>
                  </tr>
                  </tbody>
                </v-table>
              </div>
            </material-card-flat>
          </v-col>
        </v-row>
      </v-container>

      <v-container fluid class="dataTable" >
        <v-row>
          <v-col>
            <material-card
                color="blue"
                title="Listagem de Restaurante"
                text="Utilize os filtros para facilitar a busca por restaurante."
            >
              <div>
                <v-table density="compact">
                  <thead>
                  <tr>
                    <th class="text-left">
                      ID
                    </th>
                    <th class="text-left">
                      Nome
                    </th>
                    <th class="text-left">
                    </th>
                  </tr>
                  </thead>
                  <tbody>
                  <tr
                      v-for="item in restaurantes"
                      :key="item.id"
                  >
                    <td class="te">{{ item.id }}</td>
                    <td>{{ item.nome }}</td>
                    <td class="text-right">
                      <v-icon icon="mdi-dots-vertical"></v-icon>
                    </td>
                  </tr>
                  </tbody>
                </v-table>
              </div>
            </material-card>
          </v-col>
        </v-row>
      </v-container>
    </div>
</template>
<script>
import restaurantesServices from "../../services/restaurantesServices";
import Restaurante from "../../models/Restaurante";
import MaterialCardFlat from "../../components/material/CardFlat.vue";
import MaterialCard from "../../components/material/Card.vue";

export default {
  name: 'restaurante-index',
  components: {
    MaterialCardFlat,
    MaterialCard
  },
  data () {
    return {
      headers: [{
          text: 'CÃ³d',
          align: 'start',
          sortable: false,
          value: 'id',
        }],
      restaurantes: []
    }
  },
  created() {
    this.buscarListaDeRestaurantes();
  },
  methods: {
    async buscarListaDeRestaurantes() {
      const resultadoBusca = await restaurantesServices.index({});
      resultadoBusca.data.forEach((item) => {
        var restaurante = new Restaurante();
        restaurante.populate(item);
        this.restaurantes.push(restaurante);
      });
    }
  }
}
</script>