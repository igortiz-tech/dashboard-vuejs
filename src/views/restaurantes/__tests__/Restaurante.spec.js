import { describe, it, expect } from "vitest";
import { createVuetify } from "vuetify";
import { mount } from "@vue/test-utils";
import Index from "../Index.vue";


describe("Restaurante", () => {
  const vuetify = createVuetify()

  it("Existe uma pagina restaurante", () => {
    const wrapper = mount(Index,{
      global: {
        plugins: [vuetify],
      }
    });
    expect(wrapper.html()).toMatchSnapshot()
  });

  // it("Existe uma tabela", () => {
  //   const wrapper = mount(Index,{
  //     global: {
  //       plugins: [vuetify],
  //     }
  //   })
  //   const table = wrapper.find('v-table')
  //   expect(table.exists()).true;
  // });


  it("Clicar no botao", async () => {
    const wrapper = mount(Index,{
      global: {
        plugins: [vuetify],
      }
    })

    await wrapper.get('button').trigger('click');




  });

  // it("buscarRestaurante returno vazio", () => {
  //   const wrapper = mount(Index,{
  //     global: {
  //       plugins: [vuetify],
  //     }
  //   })
  //   wrapper.vm.buscarListaDeRestaurantes();
  //   expect(wrapper.vm.itens.length == 0).true;
  // });
});
