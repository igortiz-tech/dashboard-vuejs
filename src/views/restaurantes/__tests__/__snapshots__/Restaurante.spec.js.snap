// Vitest Snapshot v1

exports[`Restaurante > Existe uma pagina restaurante 1`] = `
"<div><button type=\\"button\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated\\">
    <div class=\\"v-btn__overlay\\"></div>
    <div class=\\"v-btn__underlay\\"></div>
    <!---->
    <div class=\\"v-btn__content\\" data-no-activator=\\"\\">
      <!---->
    </div>
    <!---->
    <!---->
  </button>
  <div class=\\"v-table v-theme--light v-table--density-compact\\">
    <!---->
    <div class=\\"v-table__wrapper\\">
      <table>
        <thead>
          <tr>
            <th class=\\"text-left\\"> ID </th>
            <th class=\\"text-left\\"> Nome </th>
          </tr>
        </thead>
        <tbody></tbody>
      </table>
    </div>
    <!---->
  </div>
</div>"
`;
